### PR TITLE
Feature/ts new speechrecognizer

### PR DESCRIPTION
**Pull request by alimoncul**

### CODE REVIEW

File: global/speechrecognizer/speechrecognizer.ios.ts [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+  static avaudiosession: any;
+  static avaudioengine: __SF_AVAudioEngine;
+  static speechRecognizer: __SF_SFSpeechRecognizer;
+  static speechDelegate: __SF_SFSpeechRecognizerDelegate;
+  static start(params: { locale: string; onResult: (result: any) => void; onFinish: (result: any) => void; onError: (error: SpeechRecognizerError) => void }): void {
+    SpeechRecognizerIOS.stop();
+    SpeechRecognizerIOS.onErrorHandler = params.onError;
+
+    __SF_SFSpeechRecognizer.speechRequestAuthorization(function (e) {
+      if (e.status === SFSpeechRecognizerAuthorizationStatus.authorized) {
```
Comment (cenkce) [inline global/speechrecognizer/speechrecognizer.ios.ts:37]: `SFSpeechRecognizerAuthorizationStatus` bunu enum yapip, `e` parametresinin tipinide `{status: SFSpeechRecognizerAuthorizationStatus}` yapalim

File: global/speechrecognizer/speechrecognizer.android.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  static stop(): void {
+    if (SpeechRecognizerAndroid.speechRecognizer && SpeechRecognizerAndroid.speechRecognizer.nativeObject) {
+      SpeechRecognizerAndroid.speechRecognizer.nativeObject.destroy();
+    }
+    SpeechRecognizerAndroid._isRunning = false;
+  }
+  static isRunning(): boolean {
+    return SpeechRecognizerAndroid._isRunning;
+  }
+  static createRecognizerListener(params) {
```
Comment (cenkce) [inline global/speechrecognizer/speechrecognizer.android.ts:55]: `params` in tipini verelim

File: global/speechrecognizer/speechrecognizer.ios.ts [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+import { SpeechRecognizerAndroidError, SpeechRecognizerBase, SpeechRecognizerError } from './speechrecognizer';
+
+const SFSpeechRecognizerAuthorizationStatus = {
+  notDetermined: 0,
+  denied: 1,
+  restricted: 2,
+  authorized: 3
+};
+
+function isLocaleSupport(locale) {
```
Comment (cenkce) [inline global/speechrecognizer/speechrecognizer.ios.ts:12]: `locale ` tipi `{identifier: string}`

File: global/speechrecognizer/speechrecognizer.ios.ts [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+          SpeechRecognizerIOS.avaudioengine = undefined;
+          SpeechRecognizerIOS.recognitionRequest = undefined;
+          SpeechRecognizerIOS.speechRecognizer = undefined;
+        });
+      },
+      delay: 100
+    });
+  }
+  static createRecognizer(params) {
+    if (SpeechRecognizerIOS.speechRecognizer) {
```
Comment (cenkce) [inline global/speechrecognizer/speechrecognizer.ios.ts:89]: `params`: `{locale: string}`

**Review by cenkce — CHANGES_REQUESTED** [review]

**Comment by alimoncul** [comment]
@cenkce updated

**Pull request merged**